dht/range_streamer.cc

```diff
@@ -274,7 +274,7 @@ future<> range_streamer::stream_async() {
                             nr_ranges_streamed, nr_ranges_streamed + ranges_to_stream.size(), nr_ranges_total);
                     nr_ranges_streamed += ranges_to_stream.size();
                     if (_nr_rx_added) {
-                        sp.request_ranges(source, keyspace, ranges_to_stream);
+                        sp.request_ranges(source, keyspace, std::move(ranges_to_stream));
```

Comment:
I think this is actually making the code exception unsafe (at least more unsafe than it was):
if `sp.execute().discard_result().get()` below throws, we will enter the `catch (...) { ... }` block later below, which will iterate over `ranges_to_stream` - which has now been (incorrectly) moved from.

Reply:
`std::move` does not call anything at this point, it just passes an rvalue reference down the stack.
The point in exception safety is that if an exception is thrown, the move will NOT take place.

Reply:
that's a good point.
maybe the caller needs to keep the ranges alive throughout the stream plan instead of moving them and pass a reference to the vector the the stream_plan instead, and finally consume them iff the stream_plan succeeded.

Reply:
in v7 this token range vector contains a discardable copy.
the changes to `_to_stream` are safe as they are applied only on success.